homeassistant/components/icloud/account.py

```diff
@@ -165,6 +162,10 @@ def update_devices(self) -> None:
         if self.api is None:
             return
 
+        if not self.api.is_trusted_session or self.api.requires_2fa:
```

Comment:
Is `self.api.requires_2fa` only True when reauth is required? If I recall, `self.api.requires_2sa` was always True or False

Reply:
`self.api.requires_2fa` returns `True` if Apple requires the user to enter a 2FA code, and `False` otherwise. The reason why I'm checking for it here is that if it's `True`, the API call just below to update the devices will fail.